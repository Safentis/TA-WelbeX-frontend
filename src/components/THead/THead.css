.thead {
  background-color: var(--main-dk-grey-color);
}

.thead__tr {
}

.thead__th {
  color: var(--main-dark-blue-color);
  font-size: var(--main-th-size);
  text-transform: uppercase;
}
