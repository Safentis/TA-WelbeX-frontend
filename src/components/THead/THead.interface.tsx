export interface Props {}
export type TitlesType = string[];