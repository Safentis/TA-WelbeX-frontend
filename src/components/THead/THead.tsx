import React, { FC } from 'react';

import './THead.css';
import { Props, TitlesType } from './THead.interface';

/**
 * THead is header of table component
 * @returns {React.ReactElement} - returns react element
 */
const THead: FC<Props> = ({}): React.ReactElement => {
  //* ------------------------------------------------
  //* Titles of cols
  const titles: TitlesType = ['Date', 'Name', 'Number', 'Distance'];
  const TITLES: React.ReactNode = titles.map(
    (title: string, index: number) => (
      <th className="thead__th" key={index}>
        {title}
      </th>
    ),
  );

  return (
    <thead className="thead">
      <tr className="thead__tr">{TITLES}</tr>
    </thead>
  );
};

export default THead;
