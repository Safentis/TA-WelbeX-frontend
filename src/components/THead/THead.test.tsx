import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import THead from './THead';

import { Props } from './THead.interface';

const setUp = (props: Props) => shallow(<THead {...props} />);

describe('<THead />', () => {
  let component: ShallowWrapper<
    any,
    Readonly<{}>,
    React.Component<{}, {}, any>
  >;
  let props: Props = {};

  beforeEach(() => {
    component = setUp(props);
  });

  it('component does match to the snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  describe('Markup: ', () => {
    it('has 1 thead element', () => {
      expect(component.find('thead')).toBeTruthy();
    });

    it('has 1 tr elemnt', () => {
      expect(component.find('tr')).toBeTruthy();
    });

    it('has 4 th element', () => {
      expect(component.find('th').length).toBe(4);
    });
  });
});
