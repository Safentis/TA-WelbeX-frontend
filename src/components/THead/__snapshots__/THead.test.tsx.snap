// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<THead /> component does match to the snapshot 1`] = `
<thead
  className="thead"
>
  <tr
    className="thead__tr"
  >
    <th
      className="thead__th"
      key="0"
    >
      Date
    </th>
    <th
      className="thead__th"
      key="1"
    >
      Name
    </th>
    <th
      className="thead__th"
      key="2"
    >
      Number
    </th>
    <th
      className="thead__th"
      key="3"
    >
      Distance
    </th>
  </tr>
</thead>
`;
